src/nvim/lua/executor.c

```diff
@@ -1249,11 +1249,25 @@ bool nlua_exec_file(const char *path)
 {
   lua_State *const lstate = global_lstate;
 
-  if (luaL_loadfile(lstate, path)) {
+  lua_getglobal(lstate, "loadfile");
+  lua_pushstring(lstate, path);
+
+  lua_call(lstate, 1, 2);
+
+  // loadstring() returns either:
+  //  1. nil, error
+  //  2. chunk, nil
+
+  if (lua_isnil(lstate, -2)) {
+    // 1
     nlua_error(lstate, _("E5112: Error while creating lua chunk: %.*s"));
```

Comment:
don't we need a  `lua_pop(lstate, 1);` before return to get rid of the `nil` on stack?

Reply:
I don't think so, but not 100% sure. I blindly assume the stack sorts itself out in the error cases. There are many areas where we don't clear the stack, e.g: `nlua_expand_pat`, `nlua_state_init`, `nlua_call_user_expand_func`.

Reply:
"clearing" the stack is never done, ever. This function might be called from the _global_ lua stack so it must be balanced, regardless if there is an error (which is a _vim_ error, not a lua state error) or not.